core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/coprocessor/transactional/SsccRegionEndpoint.java.tmpl

```diff
@@ -2283,6 +2578,110 @@ CoprocessorService, Coprocessor {
 
    }
 
+  public void checkAndDeleteRegionTx(RpcController controller,
+                          SsccCheckAndDeleteRegionTxRequest request,
+                          RpcCallback<SsccCheckAndDeleteRegionTxResponse> done) {
+/*
+    SsccCheckAndDeleteRegionTxResponse response = SsccCheckAndDeleteRegionTxResponse.getDefaultInstance();
+
+    byte [] rowArray = null;
+    com.google.protobuf.ByteString row = null;
+    com.google.protobuf.ByteString family = null;
+    com.google.protobuf.ByteString qualifier = null;
+    com.google.protobuf.ByteString value = null;
+    MutationProto proto = request.getDelete();
+    MutationType type = proto.getMutateType();
+    Delete delete = null;
+    Throwable t = null;
+    WrongRegionException wre = null;
+    int status = 0;
+    boolean result = false;
+    long tid = request.getTid();
+    long startId = request.getStartId();
+    boolean autoCommit = request.getAutoCommit();
+
+    java.lang.String name = ((com.google.protobuf.ByteString) request.getRegionName()).toStringUtf8();
+
+    org.apache.hadoop.hbase.coprocessor.transactional.generated.SsccRegionProtos.SsccCheckAndDeleteRegionTxResponse.Builder checkAndDeleteRegionTxResponseBuilder = SsccCheckAndDeleteRegionTxResponse.newBuilder();
+
+    if (wre == null && type == MutationType.DELETE && proto.hasRow())
+    {
+      rowArray = proto.getRow().toByteArray();
+
+      try {
+         delete = ProtobufUtil.toDelete(proto);
+      } catch (Throwable e) {
+         if (LOG.isTraceEnabled()) LOG.trace("checkAndDeleteRegionTx caught exception ", e);
+         t = e;
+      }
+
+      // Process in local memory
+      if (delete != null && t == null)
+      {
+        if (request.hasRow()) {
+          row = request.getRow();
+
+          if (!Bytes.equals(rowArray, request.getRow().toByteArray()))
+            t = new org.apache.hadoop.hbase.DoNotRetryIOException("Action's " +
+                 "Delete row must match the passed row");
+        }
+
+        if (t == null) {
```

Comment:
I know this code is presently commented out (and so, presumably a work in progress), but I'm confused by the logic that extracts the row etc. then doesn't do anything more with it. Is this just saving these values in the members of the object? If so, why do we not care about their previous state?